service/index/index_integration_test.go

```diff
@@ -256,23 +265,19 @@ func TestIndex(t *testing.T) {
 		assert.NoError(t, writer.Close())
 
 		t.Run("no types specified", func(t *testing.T) {
-			t.Parallel()
```

Comment:
Why remove `t.Parallel()` here?

Reply:
I was wondering whether or not I should put a note on this one, maybe I should have if it's not immediately obvious.

Basically since those are integration tests, the test which writes and the test(s) which read use the same database, and the parent test defers closing that database so that we don't leak. If we were to run all of the subtests in parallel, the DB would be closed immediately and they would all fail to write/read.

It could also cause them to fail since we expect the reading tests to read what the writing test wrote. The tests were already flakey since there was a chance that the read would happen before the write operation was committed.

Should I write a comment about this? If so, where, since this whole file is riddled with such tests?

Reply:
I think it would be good to have a comment for this.